hyp3_rtc_gamma/create_metadata.py

```diff
@@ -0,0 +1,239 @@
+                encoded_jpg = None
+
+            if template_suffix is not None:
+                with open("{}/RTC_GAMMA_Template{}.xml".format(etc_dir, template_suffix), "rb") as f:
+                    for line in f:
+                        line = line.replace(b"[DATE]", bytes(date, 'utf-8'))
+                        line = line.replace(b"[TIME]", bytes("{}00".format(time), 'utf-8'))
+                        line = line.replace(b"[DATETIME]", bytes(dt, 'utf-8'))
+                        line = line.replace(b"[YEARPROCESSED]", bytes("{}".format(year), 'utf-8'))
+                        line = line.replace(b"[YEARACQUIRED]", bytes(infile[17:21], 'utf-8'))
+                        line = line.replace(b"[TYPE]", bytes(input_type, 'utf-8'))
+                        line = line.replace(b"[FULL_TYPE]", bytes(full_type, 'utf-8'))
+                        line = line.replace(b"[THUMBNAIL_BINARY_STRING]", encoded_jpg)
+                        if this_pol is not None:
+                            line = line.replace(b"[POL]", bytes(this_pol, 'utf-8'))
+                        line = line.replace(b"[POWERTYPE]", bytes(power_type, 'utf-8'))
+                        line = line.replace(b"[GRAN_NAME]", bytes(granulename, 'utf-8'))
+                        line = line.replace(b"[FORMAT]", bytes(format_type, 'utf-8'))
+                        line = line.replace(b"[LOOKS]", bytes("{}".format(looks), 'utf-8'))
+                        line = line.replace(b"[FILT]", bytes("{}".format(filter_str), 'utf-8'))
+                        line = line.replace(b"[FLOOKS]", bytes("{}".format(flooks), 'utf-8'))
+                        line = line.replace(b"[SPACING]", bytes("{}".format(spacing), 'utf-8'))
+                        line = line.replace(b"[DEM]", bytes("{}".format(dem_type), 'utf-8'))
+                        line = line.replace(b"[RESA]", bytes("{}".format(resa), 'utf-8'))
+                        line = line.replace(b"[RESM]", bytes("{}".format(resm), 'utf-8'))
+                        line = line.replace(b"[HYP3_VER]", bytes("{}".format(hyp3_ver), 'utf-8'))
+                        line = line.replace(b"[GAMMA_VER]", bytes("{}".format(gamma_ver), 'utf-8'))
+                        line = line.replace(b"[TILES]", bytes("{}".format(dem_tiles), 'utf-8'))
+                        line = line.replace(b"[PCS]", bytes("{}".format(pcs), 'utf-8'))
+                        g.write(line + b'\n')
+
+    for myfile in glob.glob("*.png"):
+        with open("{}.xml".format(myfile), "wb") as g:
+            if "rgb" in myfile:
+                scale = 'color'
+                encoded_jpg = pngtothumb("{}_rgb.png".format(outfile))
+            else:
+                scale = 'grayscale'
+                encoded_jpg = pngtothumb("{}.png".format(outfile))
+
+            if "large" in myfile:
+                res = "medium"
+            else:
+                res = "low"
+
+            with open("{}/RTC_GAMMA_Template_{}_png.xml".format(etc_dir, scale), "rb") as f:
```

Comment:
This is a good use case for a jinja template

Reply:
Agreed!